floodsub.go

```diff
@@ -303,59 +351,34 @@ func (p *PubSub) publishMessage(from peer.ID, msg *pb.Message) error {
 	return nil
 }
 
-type addSub struct {
+type addSubReq struct {
 	topic string
-	sub   bool
-	resp  chan chan *Message
+	resp  chan *Subscription
 }
 
-func (p *PubSub) Subscribe(ctx context.Context, topic string) (<-chan *Message, error) {
-	return p.SubscribeComplicated(&pb.TopicDescriptor{
-		Name: proto.String(topic),
-	})
+// Subscribe returns a new Subscription for the given topic
+func (p *PubSub) Subscribe(topic string) *Subscription {
+	out := make(chan *Subscription, 1)
+	p.addSub <- &addSubReq{
+		topic: topic,
+		resp:  out,
+	}
+
+	return <-out
 }
 
 type topicReq struct {
 	resp chan []string
 }
 
+// GetTopics returns the topics this node is subscribed to
 func (p *PubSub) GetTopics() []string {
 	out := make(chan []string, 1)
 	p.getTopics <- &topicReq{resp: out}
 	return <-out
 }
 
-func (p *PubSub) SubscribeComplicated(td *pb.TopicDescriptor) (<-chan *Message, error) {
```

Comment:
We still need the topic descriptors. They arent super used yet, but thats the next step for this impl.